crates/solidity/inputs/language/src/grammar.rs

```diff
@@ -0,0 +1,883 @@
+//! Defines [`GrammarConstructorDslV2`], which allows turning the DSL v2 model into [`Grammar`]
+//! (used for generating the parser and the CST).
+
+use std::cell::OnceCell;
+use std::collections::{BTreeSet, HashMap};
+use std::rc::Rc;
+
+use codegen_grammar::Grammar;
+use codegen_grammar::GrammarElement;
+use codegen_grammar::ParserDefinition;
+use codegen_grammar::ParserDefinitionNode;
+use codegen_grammar::PrecedenceOperatorModel;
+use codegen_grammar::PrecedenceParserDefinition;
+use codegen_grammar::PrecedenceParserDefinitionNode;
+use codegen_grammar::ScannerDefinition;
+use codegen_grammar::ScannerDefinitionNode;
+use codegen_grammar::TriviaParserDefinition;
+use codegen_grammar::VersionQuality;
+use codegen_grammar::VersionQualityRange;
+use codegen_language_definition::model;
+use codegen_language_definition::model::FieldsErrorRecovery;
+use codegen_language_definition::model::Identifier;
+use codegen_language_definition::model::IndexMap;
+use codegen_language_definition::model::Item;
+
+use crate::definition::SolidityDefinition;
+
+/// Materializes the DSL v2 model ([`SolidityDefinition`]) into [`Grammar`].
+pub trait GrammarConstructorDslV2 {
```

Comment:
nit: I suggest moving this file to `codegen_grammar` crate, to keep `solidity_language` free from output-specific logic.

Reply:
I agree in general, although this still has some hacks/temporary measures like monkey-patching `SourceUnit`, directly referring to `SourceUnitMembers` and `{Leading,Trailing,EndOfFile}Trivia` as a temporary measure. I'd rather co-locate these near the actual definition, if that's okay?

I'm happy to abstract this away once the hacks are fixed :crossed_fingers:

Reply:
I think the hacks/temporary measures are fine, in order to unblock you, but do they require that this module (and `GrammarConstructorDslV2`) exist in the `solidity_language` input crate?
Not a blocker though, if you are more comfortable working with them here in the meantime.

Reply:
As I've said above, I'd rather co-locate the hack near the definition, since it refers to specific rule names, if that's okay.

After the hacks referring to specific rules are gone, I'm happy to extract it from the input crate.